lib/fog/vsphere/models/compute/datacenter.rb

```diff
@@ -24,7 +24,7 @@ def storage_pods(filters = {})
         end
 
         def vm_folders(filters = {})
-          service.folders({ datacenter: path.join('/'), type: :vm }.merge(filters))
+          service.folders({ datacenter: name, type: :vm }.merge(filters))
```

Comment:
Could we do it in all the methods, please? I guess we have the same bug in all of them.